### PR TITLE
Various commits, mostly fyi

### PR DESCRIPTION
**Pull request by elmarco**
Sorry, I lack a bit of motivation and time to write better commit messages.
Feel free to delay the merge until I improve them, or feel free to reword.

### CODE REVIEW

**Comment by whitequark** [comment]
If it's OK with you I'd prefer to wait until you've improved them.

**Comment by m-labs-homu** [comment]
:umbrella: The latest upstream changes (presumably 1ada3da5aa33917daf891dda2242d7f5beadfb10) made this pull request unmergeable. Please resolve the merge conflicts.

File: .rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ignore = [
+       "*",
```
Comment (mattkelly) [inline .rustfmt.toml]: This pattern is no longer correct, as the [format has changed to match that of `.gitignore`](https://github.com/rust-lang/rustfmt/blob/e0664662b49c1ed3e99391c51132ddf01cd87c79/Configurations.md#ignore).

The ignore list feature isn't stable yet anyway and is only available in nightly, so I suppose it [could change again](https://github.com/rust-lang/rustfmt/issues/3243).

**Comment by Dirbaio** [comment]
Extracted `add From<Address> for ::std::net::IpAddr` into #549. The other commits no longer seem to be relevant.

**Pull request closed without merging**